### ISSUE TITLE
Transform keras model to cntk model

### ISSUE THREAD

**Issue by Crisgonmu** [Microsoft/CNTK#2775]
Hi,

We are trying to change this example https://docs.microsoft.com/en-us/azure/machine-learning/preview/scenario-tdsp-biomedical-recognition?toc=%2Fen-us%2Fazure%2Fmachine-learning%2Fteam-data-science-process%2Ftoc.json&bc=%2Fen-us%2Fazure%2Fbread%2Ftoc.json to use a cnkt model instead of keras model. This is the link to the original code: https://github.com/Azure/MachineLearningSamples-BiomedicalEntityExtraction and it is divided in several sections (01_data_acquisition_and_understanding, 02_modeling, 03_deployment). We are going to focus in second section, 02_modeling and 02_model_creation part.
 
To do that, the model constructed in this way:
 
  self.model = Sequential() 
  self.model.add(Embedding(self.wordvecs.shape[0], self.wordvecs.shape[1],input_length = train_X.shape[1], weights = [self.wordvecs], trainable = False)) 
  for i in range(0, num_layers):
    if network_type == 'unidirectional':
      self.model.add(LSTM(num_hidden_units, return_sequences = True))
    else:
      self.model.add(Bidirectional(LSTM(num_hidden_units, return_sequences = True)))
    self.model.add(Dropout(dropout))
  self.model.add(TimeDistributed(Dense(train_Y.shape[2], activation='softmax')))
 
is changed for this implementation:
 
def __create_model(self, features, num_classes, num_hidden_units, dropout): 
  embedding = C.layers.Embedding(weights=self.wordvecs)(features)
  unidirectional1 = C.layers.Recurrence(C.layers.LSTM(num_hidden_units))(embedding)
  bidirectional1 = C.layers.Recurrence(C.layers.LSTM(num_hidden_units), go_backwards=True)(embedding)
  splice1 = C.splice(unidirectional1, bidirectional1)
  dropout1 = C.layers.Dropout(dropout)(splice1)
  unidirection2 = C.layers.Recurrence(C.layers.LSTM(num_hidden_units))(dropout1)
  bidirectional2 = C.layers.Recurrence(C.layers.LSTM(num_hidden_units), go_backwards=True)(dropout1)
  splice2 = C.splice(unidirection2, bidirectional2)
  dropout2 = C.layers.Dropout(dropout)(splice2)
  last = C.sequence.last(dropout2)
  model = C.layers.Dense(num_classes)(last)
  return model
 
where num_classes is equal to train_Y.shape[2] in previous code and features is C.sequence.input_variable(self.wordvecs.shape[0])
 
self.wordvecs has this shape: (66962, 50)
Train_X has this shape: (15380, 613)
Train_Y has this shape: (15380, 613, 8)
 
But we get the next error when the training is started:
The trailing dimensions of the Value shape '[0 x 613]' do not match the Variable 'Input('Input3', [#, *], [66962])' shape '[66962]'.
 
The issue have sense because the embedding layers use the weights of self.wordvec with 66962 entries but the train data has other shape, 15380 sentences with a maximun lenght of 613. However, How can we express that wordvec contains the vocabulary to use and training data has indexes to that vocabulary 
 
This link: https://github.com/Crisgonmu/EntityExtractorCNTK has the code to reproduce that issue. The model creation is in EntityExtractor.py file.

Thanks
